pkg/controller/external_tfpluginsdk.go

```diff
@@ -255,6 +256,10 @@ func (c *TerraformPluginSDKConnector) Connect(ctx context.Context, mg xpresource
 		if err != nil {
 			return nil, errors.Wrap(err, "failed to get the observation")
 		}
+		tfState, err = conversion.Convert(tfState, c.config.TFListConversionPaths(), conversion.ToSingletonList)
+		if err != nil {
+			return nil, err
```

Comment:
TODO: It would be good to wrap this error and the others below.

Reply:
Done.